### PR TITLE
Feature/autocomplete overhaul

### PR DESCRIPTION
**Pull request by HybridFox**
## PR Checklist

This PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)
- [X] A changelog entry has been added to CHANGELOG.md if necessary

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] react application / infrastructure changes
- [ ] Other... Please describe:

Issue Number: N/A

## What is the new behavior?
- `Autocomplete` Fixed items not updating when searching, added defaultValue option, added empty state, fixed open state issue
- `Flyout` Added an option to handle state change

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: packages/autocomplete/Readme.md [hunk 1]
```diff
@@ -15,7 +15,28 @@
     ]}
     onSelection={(selected) => {console.log(`Selected ${selected}`)}}
     onChange={(value) => {console.log(`Typed ${value}`)}}
-    label="Selecteer een stad">
-    </Autocomplete>
+    label="Selecteer een stad"
+></Autocomplete>
```
Comment (TriangleJuice) [inline packages/autocomplete/Readme.md]: Wij prefereren de originele manier van indenting ;-)

File: packages/flyout/src/Flyout.js [hunk 2]
```diff
@@ -45,12 +47,15 @@ class Flyout extends Component<Props> {
   }
 
   toggleIsOpen = () => {
+    const {onStateChange } = this.props;
```
Comment (TriangleJuice) [inline packages/flyout/src/Flyout.js]: Kan je je linter eens laten lopen en alle spaces en andere dingen waarover hij klaagt aanpassen. Alvast bedankt!

File: packages/autocomplete/src/Autocomplete.js [hunk 3]
```diff
@@ -24,30 +25,54 @@ class Autocomplete extends Component<Props> {
...
     };
     this.data = this.props.items;
   }
 
   static defaultProps = {
     open: false,
     onSelection: () => {},
-    onChange: () => {}
+    onChange: () => {},
+    noResultsFound: 'No results were found'
```
Comment (TriangleJuice) [inline packages/autocomplete/src/Autocomplete.js]: Als het korter kan (bv. `noResults`), graag!

File: packages/form/src/Datepicker/Datepicker.js [hunk 4]
```diff
@@ -66,7 +66,7 @@ class Datepicker extends Component<Props> {
 	}
 
 	handleClickOutside = (e) => {
-		if (this.datepicker.contains(e.target)) {
+		if (this.datepicker && this.datepicker.contains(e.target)) {
```
Comment (TriangleJuice) [inline packages/form/src/Datepicker/Datepicker.js]: Kan je deze bugfix ook mee in de CHANGELOG zetten?

File: packages/flyout/src/Flyout.js (hunk 2 above)
Comment (HybridFox) [reply packages/flyout/src/Flyout.js]: Met de default config dan? er zit geen linting config in het project :)

Comment (TriangleJuice) [reply packages/flyout/src/Flyout.js]: Oei, echt?! Ahahah…
Merci in ieder geval voor de wijzigingen!

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/),
 and this project adheres to [Semantic Versioning](https://semver.org/).
 
 ## [Unreleased]
+- `Autocomplete` Fixed items not updating when searching, added defaultValue option, added empty state, fixed open state issue
+- `Flyout` Added an option to handle state change
+- `Datepicker` Fixed an issue where the datepicker spams console when it is not loaded in correctly
```
Comment (lievenvg) [inline CHANGELOG.md]: Kleine opmerking, maar volgens mij is het gewoon `when it is not loaded correctly`.

File: packages/autocomplete/Readme.md [hunk 6]
```diff
@@ -18,4 +18,25 @@
     label="Selecteer een stad">
     </Autocomplete>
 
+The prop `defaultValue` can be used to show a default value when the component gets loaded in.
```
Comment (lievenvg) [inline packages/autocomplete/Readme.md]: Idem hier.
`when the component gets loaded.`

**Pull request merged**